site/app/views/course/CourseMaterialsView.php

```diff
@@ -287,7 +287,7 @@ private function compileAllFolderPaths(array $course_materials): array {
      * Recurses through folders and compiles an array of all the paths to folders.
      * Helper recursive function.
      *
-     * @param array<mixed> $course_materials - Dictionary: path name => CourseMaterial.
+     * @param array<string, CourseMaterial|array<string, CourseMaterial>> $course_materials - Dictionary: path name => CourseMaterial.
```

Comment:
While this results in the test being passed, the inner array type `array<string, CourseMaterial>` does not allow nested directories beyond one level. An additional test should be added and this should be corrected to use a recursive typedef.